File: src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/magic/ServerInvitesCommand.java
package com.mrpowergamerbr.loritta.commands.vanilla.magic;

import com.mrpowergamerbr.loritta.Loritta;
import com.mrpowergamerbr.loritta.LorittaLauncher;
import com.mrpowergamerbr.loritta.commands.CommandBase;
import com.mrpowergamerbr.loritta.commands.CommandCategory;
import com.mrpowergamerbr.loritta.commands.CommandContext;

import net.dv8tion.jda.core.entities.Invite;

public class ServerInvitesCommand extends CommandBase {
	@Override
	public String getLabel() {
		return "serverinvites";
	}

	@Override
	public CommandCategory getCategory() {
		return CommandCategory.MAGIC;
	}
	
	@Override
	public void run(CommandContext context) {
		if (context.getUserHandle().getId().equals(Loritta.getConfig().getOwnerId())) {
			String serverId = context.getArgs()[0];
			
			String list = "";
			for (Invite invite : LorittaLauncher.getInstance().getJda().getGuildById(serverId).getInvites().complete()) {
				list += "https://discord.gg/" + invite.getCode() + " (" + invite.getUses() + "/" + invite.getMaxUses() + ") (Criado por " + invite.getInviter().getName() + "#" + invite.getInviter().getDiscriminator() + ")\n";
			}
			context.sendMessage(context.getAsMention(true) + "\n" + list);
		} else {
			// Sem permissão
		}
	}
}

File: src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/magic/ServerInvitesCommand.kt
package com.mrpowergamerbr.loritta.commands.vanilla.magic

import com.mrpowergamerbr.loritta.LorittaLauncher
import com.mrpowergamerbr.loritta.commands.AbstractCommand
import com.mrpowergamerbr.loritta.commands.CommandCategory
import com.mrpowergamerbr.loritta.commands.CommandContext
import com.mrpowergamerbr.loritta.utils.locale.BaseLocale

class ServerInvitesCommand : AbstractCommand("serverinvites", category = CommandCategory.MAGIC, onlyOwner = true) {
	override fun getDescription(locale: BaseLocale): String {
		return "Pega os invites de um servidor a partir do ID dele"
	}

	override fun run(context: CommandContext, locale: BaseLocale) {
		val serverId = context.args[0]

		var list = ""
		for (invite in LorittaLauncher.loritta.lorittaShards.getGuildById(serverId)!!.invites.complete()) {
			list += "https://discord.gg/" + invite.code + " (" + invite.uses + "/" + invite.maxUses + ") (Criado por " + invite.inviter.name + "#" + invite.inviter.discriminator + ")\n"
		}
		context.sendMessage(context.getAsMention(true) + "\n" + list)
	}
}